pkg/cortex/serve/cortex_internal/lib/api/task.py

```diff
@@ -12,16 +12,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import os
 import imp
 import inspect
+import os
+
+import datadog
 import dill
 
-from cortex_internal.lib.exceptions import CortexException, UserException, UserRuntimeException
+from cortex_internal.lib.api.validations import validate_class_impl
+from cortex_internal.lib.exceptions import CortexException, UserException
 from cortex_internal.lib.log import configure_logger
+from cortex_internal.lib.metrics import MetricsClient
 
 logger = configure_logger("cortex", os.environ["CORTEX_LOG_CONFIG_FILE"])
 
+TASK_CLASS_VALIDATION = {
+    "required": [
+        {
+            "name": "__init__",
+            "required_args": ["self"],
+            "optional_args": ["metrics_client"],
```

Comment:
We can pass in the `metrics_client` in the `__call__` function as opposed to the `init` function. In the current design both `__init__` and `__call__` will be invoked once.